### PR TITLE
HOFF-212 add save form step

### PR DESCRIPTION
**Pull request by RobertMcCann**
What
Add a save form step for the complex form after '/select' if the user selects 'yes' an email field and a reference field appear, if the user selects 'no' the user is taken to the '/confirm' page as normal.

Why
To give users the option of saving their form

### CODE REVIEW

File: apps/demo/translations/src/en/fields.json [hunk 1]
```diff
@@ -90,5 +90,22 @@
...
+      "no": {
+        "label": "No"
+      }
+    }
+  },
+  "saveEmail": {
+    "label": "Enter your email address"
+  },
+  "saveRef": {
+    "label": "Create a memorable reference for your form",
```
Comment (al01010) [inline apps/demo/translations/src/en/fields.json:108]: What is your reference for this form?

File: apps/demo/translations/src/en/fields.json [hunk 2]
```diff
@@ -90,5 +90,22 @@
...
+    "hint": "Ensure your reference is something that is easy to remember, for example your name"
```
Comment (al01010) [inline apps/demo/translations/src/en/fields.json:109]: This is for you to keep track of your saved forms.

File: apps/demo/translations/src/en/validation.json [hunk 3]
```diff
@@ -46,5 +46,14 @@
   },
   "appealStages": {
     "required": "Select an appeal stage from the list"
+  },
+  "saveForm": {
+    "required": "Select an option below"
```
Comment (al01010) [inline apps/demo/translations/src/en/validation.json:51]: Select if you want to save your form or not.

File: apps/demo/translations/src/en/validation.json [hunk 4]
```diff
@@ -46,5 +46,14 @@
...
+  },
+  "saveEmail": {
+    "required": "Enter your email address"
+  },
+  "saveRef": {
+    "required": "Create a memorable reference"
```
Comment (al01010) [inline apps/demo/translations/src/en/validation.json:57]: You must enter a reference

File: test/_features/example_app/complex_form.feature [hunk 5]
```diff
@@ -35,8 +35,53 @@ Feature: Complex Form
...
+    Then I should be on the 'save-form' page showing 'Would you like to save your form?'
+    Then I select 'No'
+    Then I click the 'Continue' button
     Then I should be on the 'confirm' page showing 'Check your answers before submitting your application.'
     Then I click the 'Confirm submission' button
     Then I should see 'Application sent' on the page
     Then I click the 'Start again' button
     Then I should be on the 'landing-page' page showing 'Choose one of the options below and press continue.'
+
+Scenario: Complex Form Submission With Saving The Form
```
Comment (al01010) [inline test/_features/example_app/complex_form.feature:47]: Maybe add a tag to this test so it is easier to run this as a single test?

File: test/_features/example_app/validation.feature [hunk 6]
```diff
@@ -132,20 +133,23 @@ Feature: validations
...
 
-  @complex_form @continue-saved-form
+  @complex_form @continue_saved_form
   Scenario: Partial Complex Form Submission Continuing A Saved Form
     Given I start the 'base' application journey
-    Then I continue to the next step
-    Then I should see the 'Select an option below and press continue' error
     Then I choose 'Complex form'
     Then I continue to the next step
-    Then I continue to the next step
```
Comment (al01010) [inline test/_features/example_app/validation.feature:147]: Line 147 and line 148 should not be removed here

File: apps/demo/translations/src/en/fields.json (hunk 1 above)
Comment (RobertMcCann) [reply apps/demo/translations/src/en/fields.json]: changed line 108 to 'What is your reference for this form?'

File: apps/demo/translations/src/en/validation.json (hunk 3 above)
Comment (RobertMcCann) [reply apps/demo/translations/src/en/validation.json]: changed line 51 to 'Select if you want to save your form or not.'

File: apps/demo/translations/src/en/validation.json (hunk 4 above)
Comment (RobertMcCann) [reply apps/demo/translations/src/en/validation.json]: changed line 57 to 'You must enter a reference'

File: test/_features/example_app/complex_form.feature (hunk 5 above)
Comment (RobertMcCann) [reply test/_features/example_app/complex_form.feature]: added @save_form tag for this test

File: test/_features/example_app/validation.feature [hunk 7]
```diff
@@ -119,7 +119,86 @@ Feature: validations
...
     Then I click the 'Continue' button
+    Then I click the 'Continue' button
+    Then I should see the 'Select an option below' error
+    Then I select 'No'
+    Then I click the 'Continue' button
     Then I click the 'Confirm submission' button
     Then I should see 'Application sent' on the page
     Then I click the 'Start again' button
     Then I should be on the 'landing-page' page showing 'Choose one of the options below and press continue.'
+    @complex_form
```
[the remark this answers is not in the document]
Comment (RobertMcCann) [reply test/_features/example_app/validation.feature]: added @save_form tag

File: test/_features/example_app/validation.feature (hunk 6 above)
Comment (RobertMcCann) [reply test/_features/example_app/validation.feature:147]: added both lines back

**Pull request merged**